crates/examples/src/bin/batching.rs

```diff
@@ -0,0 +1,77 @@
+use log::info;
+use radix_common::math::Decimal;
+use radix_common::types::{ComponentAddress, ResourceAddress};
+use radix_common::ScryptoSbor;
+use radix_event_stream::event_handler::HandlerRegistry;
+use radix_event_stream::macros::event_handler;
+use radix_event_stream::processor::TransactionProcessor;
+use radix_event_stream::sources::gateway::GatewayTransactionStream;
+use radix_event_stream::stream::TransactionStream;
+use std::env;
+
+#[derive(Debug, Clone)]
+struct State {
+    number: u64,
+}
+
+#[derive(ScryptoSbor, Debug)]
+pub struct InstantiateEvent {
+    x_address: ResourceAddress,
+    y_address: ResourceAddress,
+    inpu_fee_rate: Decimal,
+    liquidity_pool_address: ComponentAddress,
+    pool_address: ComponentAddress,
+}
+
+#[event_handler]
+pub async fn handle_instantiate_event(
+    context: EventHandlerContext<State>,
+    event: InstantiateEvent,
+) -> Result<(), EventHandlerError> {
+    info!(
+        "Handling the {}th instantiate event: {:#?}",
+        context.state.number, event
+    );
+    context.state.number += 1;
+    Ok(())
+}
+
+#[tokio::main]
+async fn main() {
+    env::set_var("RUST_LOG", "info");
+    env_logger::init();
+
+    // Create a new handler registry
+    let mut handler_registry = HandlerRegistry::new();
+
+    // Add the instantiate event handler to the registry
+    handler_registry.add_handler(
+        "package_rdx1p5l6dp3slnh9ycd7gk700czwlck9tujn0zpdnd0efw09n2zdnn0lzx",
+        "InstantiateEvent",
+        handle_instantiate_event,
+    );
+
+    // Create a new transaction stream, which the processor will use
+    // as a source of transactions.
+    let mut gateway_stream = GatewayTransactionStream::new()
+        .from_state_version(71656472)
+        .gateway_url("https://mainnet.radixdlt.com".to_string())
+        .buffer_capacity(1000)
+        .limit_per_page(100);
+
+    let mut transactions = Vec::new();
+    let mut next = gateway_stream.start().await.unwrap();
+    // first get 10 transactions from the gateway stream
+    for _ in 0..10 {
+        let transaction = next.recv().await.unwrap();
+        transactions.push(transaction);
+    }
+
+    let mut processor =
+        TransactionProcessor::new(handler_registry, State { number: 1 });
+
+    // Process the transactions by passing in the transactions to the TransactionProcessor
+    // see also the process_transaction method to process only one transaction
+    processor.process_transactions(&transactions).await.unwrap();
```

Comment:
This looks great and is exactly what we need to write the tests in the backend 👍